modules/internal/CString.chpl

```diff
@@ -130,28 +130,54 @@ module CString {
   //
   // casts from c_string to bool types
   //
-  inline proc _cast(type t:chpl_anybool, x:c_string) throws
-    return try ((createStringWithNewBuffer(x)).strip()): t;
+  inline proc _cast(type t:chpl_anybool, x:c_string) throws {
+    var chplString: string;
+    try! {
```

Comment:
Here and in many other places this PR adds `try!` to a function that `throws`, which does not make sense to me. Even if you wanted to fix that in a 2nd PR, it would probably make sense to create that PR now by looking at this one before it's merged.

Reply:
Generally, I tried to avoid spiraling out to handle all the errors and limit the code changes in this PR. I put `try!` in `throws` function in multiple places to make sure that I don't change what that particular function can throw. Because that may mean the call sites of those function need to be changed, and if they throw their call sites and if they throw their call sites...

(Moreover, I'd expect this particular instance to turn into a try/catch where if a decode error is caught, an illegal argument is thrown from the catch block maybe by creating a junk string and attempting to cast it to bool and letting `StringCasts.chpl` throw the error. In other words, I don't think this function should ever throw a `DecodeError`)

Another reason I didn't want to do that particularly for c_strings is because they have an uncertain future.